system-x/services/google/cloud/secret-manager/src/main/java/software/tnb/google/cloud/secretmanager/service/SecretManager.java

```diff
@@ -0,0 +1,57 @@
+package software.tnb.google.cloud.secretmanager.service;
+
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.auto.service.AutoService;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+
+import com.google.cloud.secretmanager.v1.SecretManagerServiceSettings;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import software.tnb.common.service.Service;
+import software.tnb.google.cloud.common.account.GoogleCloudAccount;
+import software.tnb.google.cloud.secretmanager.validation.SecretManagerValidation;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.Base64;
+
+@AutoService(SecretManager.class)
+public class SecretManager extends Service<GoogleCloudAccount, SecretManagerServiceClient, SecretManagerValidation> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(SecretManager.class);
+
+    @Override
+    public void beforeAll(ExtensionContext extensionContext) {
+        LOG.debug("Crating Google Secret Manager validation");
+        validation = new SecretManagerValidation(client(), account().projectId());
+    }
+
+    @Override
+    public void afterAll(ExtensionContext extensionContext) {
+        if (client != null) {
+            client.close();
+        }
+    }
+
+    protected SecretManagerServiceClient client() {
+        if (client != null) {
+            return client;
+        }
+        try {
+            InputStream serviceAccountKey = new ByteArrayInputStream(Base64.getDecoder().decode(account().serviceAccountKey()));
```

Comment:
shouldn't this inputstream be closed after the client is created? :thinking:  if so, you can use try-with-resources for that